python/pyarrow/scalar.pxi

```diff
@@ -395,7 +395,7 @@ def _datetime_from_int(int64_t value, TimeUnit unit, tzinfo=None):
     dt = datetime.datetime(1970, 1, 1) + delta
     # adjust timezone if set to the datatype
     if tzinfo is not None:
-        dt = tzinfo.fromutc(dt)
+        dt = tzinfo.fromutc(dt.replace(tzinfo=tzinfo))
```

Comment:
```suggestion
        dt = dt.replace(tzinfo=datetime.timezone.utc).astimezone(tzinfo)
```

I _think_ the above results in the same, but I find that clearer / more logical to follow in the code . 
`fromutc` basically ignoring the timezone of the passed `dt` (since it interprets it as utc), while still requiring it has the same `tzinfo`, is quite confusing IMO (cfr https://bugs.launchpad.net/dateutil/+bug/943701)

Reply:
It should work the same but the change produces an error in the test (at the end).

```
>>> import pytz
>>> import datetime
>>> tzinfo = pytz.FixedOffset(120)
>>> tzinfo
pytz.FixedOffset(120)
>>> pytz.FixedOffset(120)
pytz.FixedOffset(120)
>>> 
>>> dt = datetime.datetime(2021, 1, 1)
>>> tzinfo.fromutc(dt.replace(tzinfo=tzinfo))
datetime.datetime(2021, 1, 1, 2, 0, tzinfo=pytz.FixedOffset(120))
>>> dt.replace(tzinfo=datetime.timezone.utc).astimezone(tzinfo)
datetime.datetime(2021, 1, 1, 2, 0, tzinfo=pytz.FixedOffset(120))
>>> 
>>> 
>>> arr = pa.array([0], pa.timestamp('s', tz='+02:00'))
>>> str(arr[0])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/scalar.pxi", line 89, in pyarrow.lib.Scalar.__str__
    return str(self.as_py())
  File "pyarrow/scalar.pxi", line 469, in pyarrow.lib.TimestampScalar.as_py
    return _datetime_from_int(sp.value, unit=dtype.unit(), tzinfo=tzinfo)
  File "pyarrow/scalar.pxi", line 398, in pyarrow.lib._datetime_from_int
    dt = dt.replace(tzinfo=datetime.timezone.utc).astimezone(tzinfo)
ValueError: fromutc: dt.tzinfo is not self
```

Suggest we leave it as it is.

Reply:
Sorry, my mistake! It works just as well, I'll make the change asap.